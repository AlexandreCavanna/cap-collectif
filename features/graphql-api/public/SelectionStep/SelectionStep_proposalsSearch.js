/* eslint-env jest */
const SelectionStepProposalsSearchQuery = /* GraphQL */ `
    query SelectionStepContributorsQuery($id: ID!, $terms: String, $reference: String) {
        selectionStep: node(id: $id) {
            ... on SelectionStep {
                proposals(term: $terms, reference: $reference) {
                    edges {
                        node {
                            id
                            title
                            media {
                                url(format: "reference")
                            }
                            reference
                        }
                    }
                }
            }
        }
    }
`;

describe('SelectionStep.contributors', () => {
	it('returns proposals filter by proposalTitle and reference', async () => {
		await expect(
			graphql(
				SelectionStepProposalsSearchQuery,
				{
					"terms": "Installation de bancs sur la place de la mairie",
					"reference": "1-2",
					"id": "U2VsZWN0aW9uU3RlcDpzZWxlY3Rpb25zdGVwMQ=="
				},
				'mediator',
			),
		).resolves.toMatchSnapshot();
	});
	it('returns proposals filter by proposalTitle', async () => {
		await expect(
			graphql(
				SelectionStepProposalsSearchQuery,
				{
					"terms": "Installation de bancs sur la place de la mairie",
					"id": "U2VsZWN0aW9uU3RlcDpzZWxlY3Rpb25zdGVwMQ=="
				},
				'mediator',
			),
		).resolves.toMatchSnapshot();
	});
	it('returns proposals filter by reference', async () => {
		await expect(
			graphql(
				SelectionStepProposalsSearchQuery,
				{
					"reference": "1-2",
					"id": "U2VsZWN0aW9uU3RlcDpzZWxlY3Rpb25zdGVwMQ=="
				},
				'mediator',
			),
		).resolves.toMatchSnapshot();
	});
});
