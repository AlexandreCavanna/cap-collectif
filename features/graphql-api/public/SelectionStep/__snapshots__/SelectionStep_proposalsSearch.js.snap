// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionStep.contributors returns proposals filter by proposalTitle 1`] = `
Object {
  "selectionStep": Object {
    "proposals": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "UHJvcG9zYWw6cHJvcG9zYWwz",
            "media": null,
            "reference": "1-3",
            "title": "Installation de bancs sur la place de la mairie",
          },
        },
      ],
    },
  },
}
`;

exports[`SelectionStep.contributors returns proposals filter by proposalTitle and reference 1`] = `
Object {
  "selectionStep": Object {
    "proposals": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "UHJvcG9zYWw6cHJvcG9zYWwy",
            "media": Object {
              "url": "https://capco.test/media/default/0001/01/gymnase.jpg",
            },
            "reference": "1-2",
            "title": "Rénovation du gymnase",
          },
        },
        Object {
          "node": Object {
            "id": "UHJvcG9zYWw6cHJvcG9zYWwz",
            "media": null,
            "reference": "1-3",
            "title": "Installation de bancs sur la place de la mairie",
          },
        },
      ],
    },
  },
}
`;

exports[`SelectionStep.contributors returns proposals filter by reference 1`] = `
Object {
  "selectionStep": Object {
    "proposals": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "UHJvcG9zYWw6cHJvcG9zYWwy",
            "media": Object {
              "url": "https://capco.test/media/default/0001/01/gymnase.jpg",
            },
            "reference": "1-2",
            "title": "Rénovation du gymnase",
          },
        },
      ],
    },
  },
}
`;
